src/Widget/Attribute/ModelAttributes.php

```diff
@@ -80,12 +81,12 @@ protected function getFormModel(): FormModelInterface
      *
      * @return string
      */
-    protected function getId(): string
+    protected function getId(): ?string
     {
         $new = clone $this;
 
         /** @var string */
-        $id = $new->attributes['id'] ?? $new->id;
+        $id = ArrayHelper::remove($new->attributes, 'id', $new->id);
```

Comment:
Why need remove `id` from attributes?

Reply:
Because, i am using a variable that is easier to validate than an array with `isset()`, also in all widgets i use the valid method `->id()`, instead of passing it through the attributes, there is more control.

Reply:
Confuse `clone` and modifing clonned object in getter.

Reply:
Why is it confusing?

Reply:
fixed.